projects/igniteui-angular/src/lib/list/list.component.ts

```diff
@@ -58,48 +58,147 @@ export class IgxListComponent {
 
     constructor(private element: ElementRef) {
     }
-
+    /**
+     * Returns a collection of all items and headers in the list.
+     * ```typescript
+     * let listChildren: QueryList = this.list.children;
+     * ```
+     * @memberof IgxListComponent
+     */
     @ContentChildren(forwardRef(() => IgxListItemComponent))
     public children: QueryList<IgxListItemComponent>;
-
+    /**
+     *@hidden
+     */
     @ContentChild(IgxEmptyListTemplateDirective, { read: IgxEmptyListTemplateDirective })
     public emptyListTemplate: IgxEmptyListTemplateDirective;
-
+    /**
+     *@hidden
+     */
     @ViewChild('defaultEmptyList', { read: TemplateRef })
     protected defaultEmptyListTemplate: TemplateRef<any>;
-
+    /**
+     * Sets/gets the `id` of the list.
+     * If not set, the `id` of the first list component will be `"igx-list-0"`.
+     * ```html
+     * <igx-list id = "my-first-list"></igx-list>
+     * ```
+     * ```typescript
+     * let listId =  this.list.id;
+     * ```
+     * @memberof IgxListComponent
+     */
     @HostBinding('attr.id')
     @Input()
     public id = `igx-list-${NEXT_ID++}`;
+    /**
+     * Sets/gets whether the left panning of an item is allowed.
+     * Default value is `false`.
+     * ```html
+     *  <igx-list [allowLeftPanning] = "true"></igx-list>
+     * ```
+     * ```typescript
+     * let isLeftPanningAllowed = this.list.allowLeftPanning;
+     * ```
+     * @memberof IgxListComponent
+     */
     @Input()
     public allowLeftPanning = false;
+    /**
+     * Sets/gets whether the right panning of an item is allowed.
+     * Default value is `false`.
+     * ```html
+     *  <igx-list [allowRightPanning] = "true"></igx-list>
+     * ```
+     * ```typescript
+     * let isRightPanningAllowed = this.list.allowRightPanning;
+     * ```
+     * @memberof IgxListComponent
+     */
     @Input()
     public allowRightPanning = false;
-
+    /**
+     * Emits an event with the current list when left pan gesture is executed on list item.
+     * Provides reference to the `IgxListItemComponent` as an event argument.
+     * ```html
+     * <igx-list (onLeftPan) = "onLeftPan(listItem: IgxListItemComponent)"></igx-list>
+     * ```
+     * @memberof IgxListComponent
+     */
     @Output()
     public onLeftPan = new EventEmitter<IgxListItemComponent>();
+    /**
+     * Emits an event with the current list when right pan gesture is executed on list item.
+     * Provides reference to the `IgxListItemComponent` as an event argument.
+     * ```html
+     * <igx-list (onRightPan) = "onRightPan(listItem: IgxListItemComponent)"></igx-list>
```

Comment:
The same applies here:
**<igx-list [allowRightPanning]="true" (onRightPan)="onLeftPan($event)">**